src/components/Button/Button.test.tsx

```diff
@@ -69,27 +72,71 @@ describe('Button component', () => {
     expect(onPressMock).toHaveBeenCalledTimes(1);
   });
 
-  it('Should render button component outlined', () => {
-    const button = renderButton(renderer.create, {
+  it('Should not call the given onPress function when button is disabled', () => {
+    const onPressMock = jest.fn();
+    const { queryByTestId } = renderButton(render, {
       ...defaultProps,
-      type: 'outlined',
-    }).toJSON();
+      disabled: true,
+      onPress: onPressMock,
+    });
 
-    expect(button).toMatchSnapshot();
+    fireEvent.press(queryByTestId('button'));
+
+    expect(onPressMock).not.toHaveBeenCalled();
   });
 
-  it('Should render button component contained', () => {
-    const button = renderButton(renderer.create, defaultProps).toJSON();
+  describe('variants', () => {
+    it('Should render button component outlined', () => {
+      const button = renderButton(renderer.create, {
+        ...defaultProps,
+        type: 'outlined',
+      }).toJSON();
+
+      expect(button).toMatchSnapshot();
+    });
+
+    it('Should render button component contained', () => {
+      const button = renderButton(renderer.create, defaultProps).toJSON();
+
+      expect(button).toMatchSnapshot();
+    });
+
+    it('Should render button component text', () => {
+      const button = renderButton(renderer.create, {
+        ...defaultProps,
+        disabled: true,
+        type: 'text',
+      }).toJSON();
 
-    expect(button).toMatchSnapshot();
+      expect(button).toMatchSnapshot();
+    });
   });
 
-  it('Should render button component text', () => {
-    const button = renderButton(renderer.create, {
-      ...defaultProps,
-      type: 'text',
-    }).toJSON();
+  describe('disabled variants', () => {
+    it('Should render button component outlined disabled', () => {
+      const button = renderButton(renderer.create, {
+        ...defaultProps,
+        disabled: true,
+        type: 'outlined',
+      }).toJSON();
+
+      expect(button).toMatchSnapshot();
+    });
 
-    expect(button).toMatchSnapshot();
+    it('Should render button component contained disabled', () => {
+      const button = renderButton(renderer.create, defaultProps).toJSON();
+
+      expect(button).toMatchSnapshot();
+    });
+
+    it('Should render button component text disabled', () => {
```

Comment:
É só um detalhezinho: eu costumo colocar `should` em minúsculas, porque na hora do relatório do `jest`, ele escreve bonitinho, como uma frase: `Something being tested should render...` ou até mesmo trocar o `should render` simplesmente por `renders`.